substrate-archive/src/actors/workers/blocks.rs

```diff
@@ -152,10 +152,17 @@ where
 		// ReIndexing is async process
 		let addr = ctx.address().expect("Actor just started");
 
-		let fut = ctx
-			.notify_interval(std::time::Duration::from_secs(5), || Crawl)
-			.expect("Actor cannot be disconnected; just started");
-		smol::spawn(fut).detach();
+		addr.do_send(ReIndex).expect("Actor cannot be disconnected; just started");
+
+		smol::spawn(async move {
+			loop {
+				smol::Timer::after(std::time::Duration::from_secs(5));
+				if addr.send(Crawl).await.is_err() {
+					break;
```

Comment:
After looking at this again, I don't think we need the Timer here since we await the future to finish anyway -- the purpose of the future is to just keep the blocks actor indexing.